feign-demo/src/test/java/com/nortal/test/feign/demo/glue/TestPageStepDefs.java

```diff
@@ -22,19 +22,98 @@
  */
 package com.nortal.test.feign.demo.glue;
 
+import com.nortal.test.core.services.ScenarioContext;
 import com.nortal.test.feign.demo.api.CatApi;
-import io.cucumber.java.en.Given;
-import org.springframework.beans.factory.annotation.Autowired;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpClientErrorException;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@RequiredArgsConstructor
 public class TestPageStepDefs {
-    @Autowired
-    private CatApi catApi;
+    private static final String RESPONSE = "response";
+    private static final int CURRENT_PAGE = 1;
+    private final CatApi catApi;
+    private final ScenarioContext scenarioContext;
+
+    @When("user requests cat fact")
+    public void getCatFact() {
+        try {
+            final ResponseEntity<CatApi.Fact> responseEntity = catApi.getFact();
+            scenarioContext.putStepData(RESPONSE, responseEntity);
+        } catch (HttpClientErrorException ex) {
+            scenarioContext.putStepData(RESPONSE, ex);
+        }
+    }
+
+    @Then("user should get response with cat fact")
+    public void assertCatFact() {
+        ResponseEntity<CatApi.Fact> responseEntity = scenarioContext.getStepData(RESPONSE);
+        assertNotNull(responseEntity, "Response should not be null");
+        assertEquals(HttpStatus.OK, responseEntity.getStatusCode(), "Unexpected HTTP status code");
+
+        assertNotNull(responseEntity.getBody().getFact(), "Fact should not be null");
+        assertNotNull(responseEntity.getBody().getLength(), "Length should not be null");
+    }
+
+    @When("user requests a list of cat facts")
+    public void getCatFacts() {
+        try {
+            final ResponseEntity<CatApi.FactsList> responseEntity = catApi.getFacts();
+            scenarioContext.putStepData(RESPONSE, responseEntity);
+        } catch (HttpClientErrorException ex) {
+            scenarioContext.putStepData(RESPONSE, ex);
+        }
+    }
+
+    @Then("user should get response with a list of cat facts")
+    public void assertCatFacts() {
+        ResponseEntity<CatApi.FactsList> responseEntity = scenarioContext.getStepData(RESPONSE);
+        assertNotNull(responseEntity, "Response should not be null");
```

Comment:
Assertions are being duplicated, we could put them in a different method to clean up our code